### DISCUSSION TITLE
Persist an areabrick across documents

### DISCUSSION THREAD

**Discussion by abdellahrk** [pimcore/pimcore#17405] [Q&A / Support] [answered]
Hello! Please how can I persist an Areabrick across documents? 
Let's say I create an AreaBrick sidebar and within the brick I have some Areablocks. Now if I add it (Areabrick) to my layout and add some blocks while on a particular document in the admin, it only displays on that particular page and I have to add the brick manually on all other pages.

**Comment by markus-moser** [comment, marked as answer]
You could use snippets for this (https://pimcore.com/docs/pimcore/10.6/Development_Documentation/Documents/Editables/Snippet.html). Snippets are small documents which can be embedded into other documents.

So for example you could define a header snippet, add your areabricks there and assign the snippet as (inherited) document property. Within your layout you can then read the property and if a snippet is set embedded it to the document. With the inheritance feature of the properties it's then possible to set the snippet e.g. for the whole website or a specific part in your navigation tree.

**Reply by abdellahrk** [reply]
Thank you @markus-moser. Unfortunately, I still find it difficult to grasp.

**Reply by markus-moser** [reply]
Which parts? How snippets work or how document properties work or both of them?

You could take a look at the pimcore demo implementation, there is e.g. an implementation for the footer like described:

https://github.com/pimcore/demo/blob/7ca7ffe06816a10b2f89d227e4e26c7cfbde5961/templates/layouts/layout.html.twig#L200

Or also for the sidebar:
https://github.com/pimcore/demo/blob/7ca7ffe06816a10b2f89d227e4e26c7cfbde5961/templates/layouts/layout.html.twig#L186

Inside the snippet document you can use all the editable like in a regular document, so it would be possible to use areabricks there too.

**Reply by abdellahrk** [reply]
Thank you manifold @markus-moser, it's promising and I have implemented it like in the demo project. 
Now I have one issue left. I have defined the property `footer` in the predefined properties and set that property in the root document. Unfortunately, everytime I set that property, its value disappears after refreshing the page. Hence, when I do `document.getProperty('footer')`, it returns null. Like in the picture below.
![image](https://github.com/user-attachments/assets/59e50fd5-ef47-456f-9880-10c285700003)

**Reply by markus-moser** [reply]
You need to save the document after adding the property. Did you do this?

**Reply by abdellahrk** [reply]
Yes. I always save it @markus-moser. Even though the property is from a predefined property settings, it loses its value always at the document level.
Another thing is that I cannot set the value of a property when I create one on a document if the type is document.

**Reply by markus-moser** [reply]
This sounds really strange. I never had a similar problem. Which Pimcore + Admin Ui Classic Bundle version do you use?

**Reply by abdellahrk** [reply]
Sorry for the very late reply, @markus-moser. 
I am using Pimcore 11.2.6
and admin UI Classic 1.4.4

**Reply by markus-moser** [reply]
That's strange. I cannot find any relevant bugfixes since this version on for me it works perfectly in the current version. I also never had a problem with this functionallity (which I used for many years in different versions). In your screenshot it seems that it works for the faqs_snippet but not for the footer? Is there any kind of difference?

**Reply by abdellahrk** [reply]
Yes. There's a difference. The footer I was trying to use a partial document including it with `pimcore_inc` and rendering it like a normal document. What you pointed out to me to do and I have seen that, it's been done with the pimcore demo. I finally turned my footer to a snippet and it's working.
Thank you manifold @markus-moser
